유저 회원가입/로그인 기능

Zoe0631 opened the pull request:
#### 0614 update

1. 회원가입 유효성 검사 추가 완료.
2. ip 대역 검사는 보류(천천히 공부하면서 진행할 예정) #5 
~~3. 로그인 api 작성 필요~~
4. hubert님과의 대화가 더 필요하다고 생각되는것 외에는 피드백 반영. resolve 안된 내용에 대해서 피드백 덧붙여주시면 감사하겠습니다. 🙏

#### 0623 update

1.  hierarchy 구조로 예외 처리 리팩토링
2. 로그인(signIn)/로그아웃(signOut)/리소스 요청 시 로그인 했는지 여부 체크(SignInCheckfilter)


------------------------------

 user 기능 중 해야할 것
 - 인증번호 저장 redis에 하기 #4 
 - ip 대역확인 #5 

위의 기능들은 +@ 같은 느낌이라서 우선 해당프로젝트의 주요기능인 결제기능 코드 작성을 완료한 뒤, 혹은 결제기능 작성하면서 짬짬히 해갈 생각입니다.

Zoe0631 committed:
feat : add signUpUser Api frame

TODO : add check ip range and validation

Zoe0631 committed:
build : add dependency for user domain

Zoe0631 committed:
refactor : add check ip cidr frame

need to add below list
- wrapping request body for using request object at interceptor
- get Korea ip range list

Zoe0631 committed:
refactor : add @Componentscan at container config

f-lab-hubert reviewed (commented):
소소한 코멘트 달아보았습니다 🙏 여러가지 이야기해볼게 많지만 확인해주시면 이어서 리뷰를 드려볼게요 :)

Zoe0631 committed:
refactor : reflecting pr feedback

1. build.gradle annotation for spring boot init
2. delete unnecessary annotations(@Componentscan)
3. using ThreadLocalRandom instead of Random
4. change core/data/BaseEntity
5. change core/utils/IpUtil

Zoe0631 committed:
refactor : update application.yml for gracefulshutdown

Zoe0631 committed:
refactor : update VerificationEmail API

1. add Exception when user email already exist
2. change naming eMail to email

Zoe0631 committed:
refactor : update BaseEntity

compatible timezone and store data row

Zoe0631 committed:
build : add dependency for validation

Zoe0631 committed:
refactor : resolve timezone issue

change db timezone and baseentity type to OffsetDateTime for multinational service

Zoe0631 committed:
feat : add validation check for user sign up

Zoe0631 committed:
refactor : update user service

1. check is user already sign up (sign up api)
2. change dto name for assign one dto for one api
3. check when find user verifiaction entity by verificationId, email info accord

Zoe0631 committed:
chore : line breaks, delete unused dependency

Zoe0631 committed:
feat : add signIn and signOut api

Zoe0631 committed:
chore : delete annotation for description of code

Zoe0631 committed:
refactor : refactoring of feat

1. separate function comparePasswordAndConfirmPassword
2. when user is already sign up, throw Exception in verifyUserEmail function

Zoe0631 committed:
refactor : hierarchical exception handling

Zoe0631 committed:
chore : withhold function that check ip range feat

Zoe0631 committed:
chore : remove unnecessary code

f-lab-hubert reviewed (commented):
제 리뷰를 모두 반영해서 develop에 머지하지않으셔도 됩니다. 그렇게되면 프로젝트가 끝날 때까지 머지를 할 수 없을 것 같아요. 적절하게 피드백을 반영하셨으면 develop에 머지후 새로운 브랜치로 작업하시면 좋을 것 같습니다. 🙏

Zoe0631 committed:
refactor : reflecting feedback

Zoe0631 committed:
refactor : verify email by redis instead of mysql

Zoe0631 committed:
chore : unnecessary somethings

Zoe0631 committed:
Merge pull request #6 from f-lab-edu/feature/4

유저 인증번호 rdb 대신 redis 에 저장하기
